src/main/kotlin/com/hibob/academy/resource/PetApi.kt

```diff
@@ -0,0 +1,81 @@
+package com.hibob.academy.resource
+
+import com.hibob.restAPI.Pet
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import org.springframework.stereotype.Controller
+import java.time.LocalDate
+import java.util.concurrent.CopyOnWriteArrayList
+
+@Controller
+@Path("/api/adi/pets")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+class PetsResource {
+
+    private val allPets: MutableList<Pet> = CopyOnWriteArrayList()
+
+    @GET
+    @Path("/{petId}")
+    fun getPetById(@PathParam("petId") petId: Long): Response {
+        val pet = allPets.find { it.id == petId.toInt() }
+        if (petId.toInt() == 123)
+            throw NotFoundException("Pet with id $petId not found")
+
+        return if (pet != null)
```

Comment:
The not found is already thrown in an exception so the if else in this case is doing the same and not needed.

Reply:
Do you mean I should throw an exception instead of make the response?
Because when I get the Exception I get HTTP 404 without any message in Postman